### PR TITLE
chore: ran eslint-plugin-jsdoc/check-tag-names autofixer

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
**Context**: Part of part of #648, porting old [TSLint](https://palantir.github.io/tslint) rules to [ESLint](https://eslint.org/) equivalents. Within migrating `no-redundant-jsdoc2`, this runs the auto-fixer from [jsdoc/check-tag-names](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/docs/rules/check-tag-names.md#user-content-check-tag-names-options-typed).

`jsdoc/check-tag-names` reports on JSDoc tags that are unnecessary. Here, we're using its `{ typed: true }` option to look for JSDoc tags that are redundant in TypeScript.

<details>
<summary>I used a <em>very</em> permissive ESLint config to avoid flagging library-specific tags or switching tag names unnecessarily (e.g. <code>@return</code> to <code>@returns</code>)</summary>

```json
{
    "root": true,
    "extends": [],
    "parser": "@typescript-eslint/parser",
    "plugins": ["@typescript-eslint", "jsdoc"],
    "rules": {
      "jsdoc/check-tag-names": [
        "error",
        {
          "definedTags": [
            "addVersion",
            "api",
            "author",
            "beta",
            "brief",
            "category",
            "cfg",
            "chainable",
            "check",
            "classDescription",
            "condparamprivilege",
            "constraint",
            "credits",
            "declaration",
            "defApiFeature",
            "defaultValue",
            "detail",
            "end",
            "eventproperty",
            "experimental",
            "export",
            "expose",
            "extendscript",
            "factory",
            "field",
            "final",
            "fixme",
            "fluent",
            "for",
            "governance",
            "header",
            "hidden-property",
            "hidden",
            "id",
            "label",
            "language",
            "link",
            "listen",
            "locus",
            "methodOf",
            "minVersion",
            "ngdoc",
            "nonstandard",
            "note",
            "npm",
            "observable",
            "option",
            "optionobject",
            "options",
            "packageDocumentation",
            "param",
            "parent",
            "platform",
            "plugin",
            "preserve",
            "privateRemarks",
            "privilegeLevel",
            "privilegeName",
            "proposed",
            "range",
            "readOnly",
            "related",
            "remark",
            "remarks",
            "required",
            "requires",
            "restriction",
            "returnType",
            "section",
            "see",
            "since",
            "const",
            "singleton",
            "source",
            "struct",
            "suppress",
            "targetfolder",
            "enum",
            "title",
            "record",
            "title",
            "TODO",
            "trigger",
            "triggers",
            "typeparam",
            "typeParam",
            "unsupported",
            "url",
            "usage",
            "warn",
            "warning",
            "version"
          ],
          "typed": true
        }
      ]
    },
    "settings": {
        "jsdoc": {
            "tagNamePreference": {
                "argument": "argument",
                "exception": "exception",
                "function": "function",
                "method": "method",
                "param": "param",
                "return": "return",
                "returns": "returns"
            }
        }
    }
  }
```

</details>

Starting off as a draft so we can discuss what changes might need to be made to `check-tag-names`. Each classification of change has a comment thread inline.

### CODE REVIEW

File: types/angular-ui-bootstrap/index.d.ts
Comment (JoshuaKGoldberg) [inline types/angular-ui-bootstrap/index.d.ts:513]: Removing `@type` is the most common change in the rule. This instance is a good exemplar - the type is inaccurate!

File: types/asana/index.d.ts
Comment (JoshuaKGoldberg) [inline types/asana/index.d.ts:20]: `@class` is unnecessary when decorating a `new` signature. It's implied by it being `new`.

File: types/asana/index.d.ts
Comment (JoshuaKGoldberg) [inline types/asana/index.d.ts:385]: `@constructor` is unnecessary when decorating a `new` signature. It's implied by it being `new`. See also `@class`.

File: types/bigint/index.d.ts
Comment (JoshuaKGoldberg) [inline types/bigint/index.d.ts:355]: This one I'm wondering about... why is this documented publicly if it's meant to be an internal function? Should it be removed from the types?

My personal instinct as someone who has never used `@types/bigint` would be to lean towards removing it. cc @Evgenus - are you up for discussing here so we can keep it in the context of `jsdoc/check-tag-names`?

File: types/chrome-apps/index.d.ts
Comment (JoshuaKGoldberg) [inline types/chrome-apps/index.d.ts:6342]: As a feature request for the autofixer, it'd be nice if these could be moved into JSDoc tags specific to the properties...

File: types/documentdb-server/documentdb-server-tests.ts
Comment (JoshuaKGoldberg) [inline types/documentdb-server/documentdb-server-tests.ts:262]: `@function` is unnecessary when describing something typed as a function.

File: types/dwt/Addon.OCR.d.ts
Comment (JoshuaKGoldberg) [inline types/dwt/Addon.OCR.d.ts:10]: Hmm, also unintentional, should be reverted.

```suggestion
     * @argument errorCode The error code.
     * @argument errorString The error string.
```

File: types/openfin/_v2/api/external-window/external-window.d.ts
Comment (sandersn) [inline types/openfin/_v2/api/external-window/external-window.d.ts:63]: another weird change. I've never heard of either of these tags.

File: types/openfin/_v2/api/platform/platform-provider.d.ts
Comment (sandersn) [inline types/openfin/_v2/api/platform/platform-provider.d.ts:6]: all these typedefs should be deleted, or if they exist as real type aliases elsewhere, moved there

File: types/openlayers/v3/index.d.ts
Comment (sandersn) [inline types/openlayers/v3/index.d.ts:68]: this `implements` is missing from the type

File: types/selenium-webdriver/remote.d.ts
Comment (sandersn) [inline types/selenium-webdriver/remote.d.ts:78]: these ought to be deleted too

**Review by sandersn — COMMENTED** [review]
I like this, and left some comments in places that could improve. (Although any improvement is welcome.)

File: types/googlemaps.infobubble/index.d.ts
Comment (JoshuaKGoldberg) [reply types/googlemaps.infobubble/index.d.ts:13]: Yeah. `@fileoverview` is a synonym for it: https://jsdoc.app/tags-file.html

File: types/matter-js/index.d.ts
Comment (JoshuaKGoldberg) [reply types/matter-js/index.d.ts:3618]: Hmm yeah. I can revert these for now and tackle them later...

File: types/next-pwa/index.d.ts
Comment (JoshuaKGoldberg) [reply types/next-pwa/index.d.ts:20]: https://jsdoc.app/tags-abstract.html believes it to be the preferred tag, with `@virtual` as a synonym. I'll just revert - these should be ignored.

File: types/openlayers/v3/index.d.ts
Comment (JoshuaKGoldberg) [reply types/openlayers/v3/index.d.ts]: I ended up reverting this back to `master`. +1 to fixing it up, but I'm running low on time budget...

File: types/selenium-webdriver/remote.d.ts
Comment (JoshuaKGoldberg) [reply types/selenium-webdriver/remote.d.ts]: I ended up reverting this back to `master`. +1 to fixing it up, but I'm running low on time budget...

File: types/openfin/_v2/api/platform/platform-provider.d.ts
Comment (JoshuaKGoldberg) [reply types/openfin/_v2/api/platform/platform-provider.d.ts]: I ended up reverting this back to `master`. +1 to fixing it up, but I'm running low on time budget...

File: types/openfin/_v2/api/external-window/external-window.d.ts
Comment (JoshuaKGoldberg) [reply types/openfin/_v2/api/external-window/external-window.d.ts]: Yeah I ended up reverting this back to `master`.

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## angular-ui-bootstrap ([<kbd>unpkg</kbd>](https://unpkg.com/browse/angular-ui-bootstrap@latest/))
was missing the following properties:
1. padStart
1. padEnd
1. trimLeft
1. trimRight
1. trimStart

<details>
<summary>as well as these 1 other properties...</summary>
<p>trimEnd</p>
</details>

## confidence ([<kbd>unpkg</kbd>](https://unpkg.com/browse/confidence@latest/))
was missing the following properties:
1. id
## express-brute-memcached ([<kbd>unpkg</kbd>](https://unpkg.com/browse/express-brute-memcached@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'express-brute-memcached'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
1. defaults
1. The declaration doesn't match the JavaScript module 'express-brute-memcached'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
1. defaults
## express-brute ([<kbd>unpkg</kbd>](https://unpkg.com/browse/express-brute@latest/))
was missing the following properties:
1. defaults
1. instanceCount
## figlet ([<kbd>unpkg</kbd>](https://unpkg.com/browse/figlet@latest/))
was missing the following properties:
1. preloadFonts
1. figFonts
## firebase-token-generator ([<kbd>unpkg</kbd>](https://unpkg.com/browse/firebase-token-generator@latest/))
was missing the following properties:
1. isEmptyObject_
1. isUselessOptionsObject_
## jsts ([<kbd>unpkg</kbd>](https://unpkg.com/browse/jsts@latest/))
was missing the following properties:
1. abstractFunc
1. error
1. index
1. geomgraph
1. noding

<details>
<summary>as well as these 1 other properties...</summary>
<p>triangulate</p>
</details>

## lokijs ([<kbd>unpkg</kbd>](https://unpkg.com/browse/lokijs@latest/))
was missing the following properties:
1. deepFreeze
1. freeze
1. unFreeze
1. Resultset
1. Comparators
## loopback ([<kbd>unpkg</kbd>](https://unpkg.com/browse/loopback@latest/))
was missing the following properties:
1. version
1. registry
1. errorHandler
1. faviconFile
1. remoteMethod

<details>
<summary>as well as these 15 other properties...</summary>
<p>template, createModel, configureModel, findModel, getModel, getModelByType, createDataSource, memory, DataSource, Connector, Memory, Mail, Remote, GeoPoint, ValidationError</p>
</details>

## memwatch-next ([<kbd>unpkg</kbd>](https://unpkg.com/browse/memwatch-next@latest/))
was missing the following properties:
1. addListener
1. once
1. removeListener
1. off
1. removeAllListeners

<details>
<summary>as well as these 10 other properties...</summary>
<p>setMaxListeners, getMaxListeners, listeners, rawListeners, emit, listenerCount, prependListener, prependOnceListener, eventNames, gc</p>
</details>

## ng-file-upload ([<kbd>unpkg</kbd>](https://unpkg.com/browse/ng-file-upload@latest/))
was missing the following properties:
1. padStart
1. padEnd
1. trimLeft
1. trimRight
1. trimStart

<details>
<summary>as well as these 1 other properties...</summary>
<p>trimEnd</p>
</details>

## nodal ([<kbd>unpkg</kbd>](https://unpkg.com/browse/nodal@latest/))
was missing the following properties:
1. mocha
1. include
1. env
## onfleet__node-onfleet ([<kbd>unpkg</kbd>](https://unpkg.com/browse/onfleet__node-onfleet@latest/))
was missing the following properties:
1. initBottleneckOptions
## page ([<kbd>unpkg</kbd>](https://unpkg.com/browse/page@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'page'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
1. callbacks
1. exits
1. base
1. strict

<details>
<summary>as well as these 16 other properties...</summary>
<p>start, stop, show, back, redirect, replace, dispatch, exit, configure, sameOrigin, clickHandler, create, len, current, Context, Route</p>
</details>

## restify ([<kbd>unpkg</kbd>](https://unpkg.com/browse/restify@latest/))
was missing the following properties:
1. helpers
## sat ([<kbd>unpkg</kbd>](https://unpkg.com/browse/sat@latest/))
was missing the following properties:
1. V
1. isSeparatingAxis
## selenium-webdriver ([<kbd>unpkg</kbd>](https://unpkg.com/browse/selenium-webdriver@latest/))
was missing the following properties:
1. ThenableWebDriver
## sequelize ([<kbd>unpkg</kbd>](https://unpkg.com/browse/sequelize@latest/))
was missing the following properties:
1. version
1. options
1. DataTypes
1. Association
1. useInflection
## simplesmtp ([<kbd>unpkg</kbd>](https://unpkg.com/browse/simplesmtp@latest/))
was missing the following properties:
1. createServer
1. connect
1. createClientPool
1. version
## tuya-panel-kit ([<kbd>unpkg</kbd>](https://unpkg.com/browse/tuya-panel-kit@latest/))
was missing the following properties:
1. Drawer
1. Wave
1. Diffusion
1. OfflineView
1. FullView
## twitter-stream-channels ([<kbd>unpkg</kbd>](https://unpkg.com/browse/twitter-stream-channels@latest/))
was missing the following properties:
1. getMockedClass
1. launchMockDataRetriever
## winjs ([<kbd>unpkg</kbd>](https://unpkg.com/browse/winjs@latest/))
was missing the following properties:
1. ColorTypes
1. createAccentRule
1. Pivot
1. LightDismissalReasons
1. DismissalPolicies

<details>
<summary>as well as these 40 other properties...</summary>
<p>LightDismissableElement, ModalElement, shown, hidden, updated, isShown, isTopmost, keyDown, keyUp, keyPress, ClassNames, EventNames, actionAreaCommandWidth, actionAreaSeparatorWidth, actionAreaOverflowButtonWidth, overflowCommandHeight, overflowSeparatorHeight, controlMinWidth, overflowAreaMaxWidth, heightOfMinimal, heightOfCompact, contentMenuCommandDefaultLabel, defaultClosedDisplayMode, defaultOpened, defaultOverflowDirection, typeSeparator, typeContent, typeButton, typeToggle, typeFlyout, commandSelector, primaryCommandSection, secondaryCommandSection, OverflowDirection, OpenCloseMachine, ToolBar, SplitView, SplitViewPaneToggle, defaultPlacement, AppBar</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against dace4a1664f106f716aac6c127a16f7752e4d0af
</p>

File: types/bigint/index.d.ts
Comment (JoshuaKGoldberg) [reply types/bigint/index.d.ts:355]: Great! I'm adding in a `@note` to make this more in line with JSDoc, since it kind of is a "note" / not part of the description.

File: types/openfin/_v2/api/clipboard/clipboard.d.ts
Comment (JoshuaKGoldberg) [reply types/openfin/_v2/api/clipboard/clipboard.d.ts:7]: 🤔 interesting, there's already a `WriteRequestType` interface there... I'll just delete and let the author review if possible.

**Comment by JoshuaKGoldberg** [comment]
@sandersn I think the code changes are ready for review from authors. I'm hesitant to open a PR that touches almost ~200 packages - and also hesitant to open almost ~200 PRs 😂. Is there a way you'd want me to request review?

**Comment by sandersn** [comment]
I normally split everything into 1 p For relatively uncontroversial and required changes that touch *everything*, 1 PR is fine and I just wait for a review by a DT maintainer.

This change is not *required* but I hope it's not controversial. So I think a single PR is fine. You might split out fibjs separately since it has so many files.

File: types/chrome-apps/index.d.ts
Comment (JoshuaKGoldberg) [reply types/chrome-apps/index.d.ts:6341]: https://github.com/gajus/eslint-plugin-jsdoc/issues/1054

**Review by gabritto — APPROVED** [review]

**Comment by JoshuaKGoldberg** [comment]
I ... guess I'll just go ahead then! 😄 

Ready to merge

**Comment by JoshuaKGoldberg** [comment]
Ready to merge

**Pull request merged**
